tests/ui/cypress/specs/multi_user_openid_connect.js

```diff
@@ -0,0 +1,289 @@
+          cy.contains('return to the login page')
+        } else if (ts.d == 'badidtoken') {
+          cy.contains('OpenID Connect: Code exchange failed: Failed to parse server response')
+          cy.contains('return to the login page')
+        } else if (ts.d == 'badrole') {
+          cy.contains(
+            'Your user does not have sufficient rights to perform this action. Please contact your administrator.'
+          )
+          cy.contains('return to the login page')
+        }
+      }
+    )
+  })
+
+  it('Can logout', () => {
+    // login
+    cy.visit('/')
+    cy.url().should('not.include', Cypress.config('baseUrl'))
+    cy.contains('Mock OpenID Connect login form')
+    cy.get('input[name="username"]').clear().type(admin.u)
+    cy.contains('Sign In').click()
+
+    // verify that we are shown to be logged in to the Krill UI
+    cy.contains('Sign In').should('not.exist')
+    cy.url().should('include', Cypress.config('baseUrl'))
+    cy.get('#userinfo').click()
+    cy.get('#userinfo_table').contains(admin.u)
+
+    // logout
+    cy.intercept('GET', /^http:\/\/localhost:1818\/logout.+/).as('oidcLogout')
+    cy.get('.logout').click()
+    cy.wait('@oidcLogout').its('response.statusCode').should('eq', 302)
+
+    // verify that we are shown the OpenID Connect provider login page
+    cy.url().should('not.include', Cypress.config('baseUrl'))
+    cy.contains('Mock OpenID Connect login form')
+    cy.get('input[name="username"]')
+  })
+
+  it('Login with short-lived non-refreshable token and try to refresh page', () => {
+    cy.intercept('GET', '/api/v1/authorized').as('isAuthorized')
+    cy.visit('/')
+
+    cy.wait('@isAuthorized').its('response.statusCode').should('eq', 403)
+    cy.url().should('not.include', Cypress.config('baseUrl'))
+    cy.contains('Mock OpenID Connect login form')
+    cy.get('input[name="username"]').clear().type(shorttoken.u)
+
+    cy.intercept('GET', '/index.html').as('postLoginIndexFetch')
+    cy.intercept('GET', '/api/v1/authorized').as('isAuthorized')
+    cy.contains('Sign In').click()
+
+    cy.wait('@postLoginIndexFetch').its('response.statusCode').should('eq', 200)
+    cy.wait('@isAuthorized').its('response.statusCode').should('eq', 200)
+    cy.url().should('include', Cypress.config('baseUrl'))
+    cy.contains('Sign In').should('not.exist')
+    cy.get('#userinfo').click()
+    cy.get('#userinfo_table').contains(shorttoken.u)
+    cy.contains(shorttoken.u)
+    cy.contains('Welcome to Krill')
+
+    // the token has a lifetime of 5 second and no refresh token
+    // wait 6 seconds...
+    // note: a shorter token with a 1 second lifetime doesn't work in the GitHub
+    // Action runner environment because the token has sometimes already expired
+    // by the time Krill verifies it!
+    cy.wait(6000)
+
+    // Try to create a CA, by typing in the input, clicking the 'Create CA' button
+    // and then clicking 'Ok'. This should fail, since the token can't be refereshed.
```

Comment:
Minor improvement suggestion: move `[valid refresh]` to the end of the test name otherwise when viewing the list of tests all of the other test names start with an English description but these tests do not which makes the list of test names harder to read (for me at least).

Reply:
Do you want all the bracketed usernames at the end? I put them in front because the username is what makes them discernable.

Reply:
If you don't immediately think it would be better the other way I'd say leave it as you have a reason for the naming strategy you chose and the most important thing is the test itself, I can live with the name.

Reply:
Maybe instead of `[${ts.u}] Login with short-lived non-refreshable token and try to create a CA` do \[${ts.u}\] `Login as ${ts.u} and try to create a CA` ? It's more concise and better for humans?